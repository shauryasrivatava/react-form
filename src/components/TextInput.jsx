import React from 'react';
import PropTypes from 'prop-types';

const TextInput = (props) => {
    return (
        <div className="form-group">
            <label htmlFor={props.labelFor}>{props.label}</label>
            <input type={props.type} name={props.name} id={props.id} value={props.value} onChange={props.handleChange} className={props.className} />      
        </div>
    )
}

TextInput.propTypes = {
    label: PropTypes.string,
    labelFor: PropTypes.string,
    value: PropTypes.string,
    handleChange: PropTypes.func,
    id: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.string,
    className: PropTypes.string
};

TextInput.defaultProps = {
    label: 'Text Input',
    labelFor: 'text input',
    value: 'Hello Text Input',
    handleChange: function(){},
    id: 'input',
    name: 'input',
    type: 'input',
    className: 'form-control'
}

export default TextInput;