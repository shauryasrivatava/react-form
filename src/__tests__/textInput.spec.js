import renderer from 'react-test-renderer';
import TextInput from '../components/TextInput';
import React from 'react';
import { shallow } from 'enzyme';
import enzymeConfig from '../../enzymeConfig';

describe('<TextInput />', function() {

    it('Should render correctly', function(){
        
        const component = renderer.create(<TextInput />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should have <div> with correct class name', function(){

        const component = shallow(<TextInput />);
        expect(component.find('div').hasClass('form-group')).toBe(true);
    });

    it('Should have <label> with default label title', function(){

        const component = shallow(<TextInput />);
        expect(component.find('label').text()).toEqual('Text Input');
    });

    it('Should call changeHandler onChange on input', function() {

        const mockHandler = jest.fn();
        const component = shallow(<TextInput handleChange={mockHandler}/>);
        const input = component.find('input');
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);

    })
})