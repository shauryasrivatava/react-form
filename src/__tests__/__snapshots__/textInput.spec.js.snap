// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> Should render correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="text input"
  >
    Text Input
  </label>
  <input
    className="form-control"
    id="input"
    name="input"
    onChange={[Function]}
    type="input"
    value="Hello Text Input"
  />
</div>
`;
